### ISSUE TITLE
how to take screenshot on failure step, not in afterEach method

### ISSUE THREAD

**Issue by anudeepsinghpatelgithub** [Evilweed/protractor-beautiful-reporter#61]
I am always getting screentshot of the page which is defind in afterEach method. 

how to take screenshot on failure step, not in afterEach method. below is my sample code

```
  beforeAll(function () {
    jasmine.DEFAULT_TIMEOUT_INTERVAL = protractorConf.timeouts.extendedDefaultTimeoutInterval;
  });

  beforeEach(function (done) {
    setTimeout(function () {
      browser.waitForAngular();

      expect(browser.getCurrentUrl()).toContain('/login');
      done();
    }, protractorConf.timeouts.pageResolveTimeout);
  });

  afterEach(function (done) {
    setTimeout(function () {
      browser.waitForAngular();
      page.main.doAccountAction('logout');//always screenshot of this page is stored, doesn't matter whether test is pass or fail
      done();
    }, protractorConf.timeouts.pageResolveTimeout);
  });

  afterAll(function () {
    browser.executeScript('window.ss.clear();');
    browser.executeScript('window.ss.clear();');
  });

  using(dataProvider, function (data, description) {
    it('should check ' + description, () => {
        loginPage.loginAs(data.user);
        navigateToDefaultSubCategory();

        // Validate presence of Create Button
        element(page.analysisElems.addAnalysisBtn.isDisplayed().then(function (isVisible) {
          expect(isVisible).toBe(data.create,
            "Create button expected to be " + data.create + " on Analyze Page, but was " + !data.create);
        }));
})
});
```

**Comment by miller45** [comment]
I am not sure how to solve this...but have you tried to do the logout in the beforeEach step?
Or else not await (done()) the logout step in afterEach?

**Comment by miller45** [comment]
Closing ...merged the request in #102
